feat(react-native): Add Expo as a guide under React Native (#18476)
getsentry/sentry-docs: docs/platforms/react-native/guides/expo/index.mdx

---
title: Expo
description: "Learn how to set up an Expo project with Sentry's React Native SDK, including the Sentry Expo config plugin, Expo Router tracing, and Expo Updates context."
caseStyle: camelCase
supportLevel: production
sdk: sentry.javascript.react-native
fallbackPlatform: react-native
categories:
  - mobile
---

<AgentSkillsCallout skill="sentry-react-native-sdk" platformName="Expo" />

> On this page, we get you up and running with Sentry's React Native SDK in an [Expo](https://expo.dev/) project. The same `@sentry/react-native` SDK powers both bare React Native and Expo apps — this guide covers the Expo-specific setup (config plugin, Metro, EAS) and integrations.

Reviewer: ```suggestion
Use this guide to get fully set up with Sentry's React Native SDK in an [Expo](https://expo.dev/) project. The same `@sentry/react-native` SDK powers both bare React Native and Expo apps. This guide covers the Expo-specific setup (config plugin, Metro, EAS) and integrations.
```

Author: Applied with f9686c4c2

ps. Sorry for not applying your suggestion directly but it wasn't possible after merging from `master`

<Alert>

If you don't already have an account and Sentry project established, head over to [sentry.io](https://sentry.io/signup/), then return to this page.

</Alert>

## Prerequisites

- [Expo SDK 50](https://docs.expo.dev/workflow/upgrading-expo-sdk-walkthrough/) or newer.

>   - [Expo SDK 49](https://docs.expo.dev/guides/using-sentry/) and older are supported by the `sentry-expo` package.

Reviewer: Do we have more information about the sentry-expo package? This feels like a dead end otherwise.

Reviewer: And are we also saying that you can't use the native Expo SDK unless you migrate from it/update to Expo SDK 50? It sounds like that, but want to double check.

Author: Good catch 👍 Updated with f9686c4c2

- [Migrate from `sentry-expo` to `@sentry/react-native`](/platforms/react-native/migration/sentry-expo/).
- [Sign up for an account](https://sentry.io/signup/).

## Automatic Install

> Use the [Sentry Wizard](https://github.com/getsentry/sentry-wizard) to patch your project automatically, as shown below. Alternatively, you can follow the [Manual Install](#manual-install) if you prefer. You only need to patch the project once. Then, you can add the patched files to your version control system.

Reviewer: Do we need to link to the wizard repo? I feel like it might be confusing - like go to the repo to install? Versus, here's the command in terminal (which is right below). But you tell me if this addition is valuable. 

```suggestion
Use the Sentry Wizard to patch your project automatically, as shown below. Alternatively, you can follow the [Manual Install](#manual-install) if you prefer. You only need to patch the project once. Then, you can add the patched files to your version control system.
```

Author: Updated with f9686c4c2

  // ___PRODUCT_OPTION_START___ logs
  // Enable logs to be sent to Sentry
  // Learn more at https://docs.sentry.io/platforms/react-native/logs/
  enableLogs: true,
  // ___PRODUCT_OPTION_END___ logs
  // ___PRODUCT_OPTION_START___ profiling
  // profilesSampleRate is relative to tracesSampleRate.
  // Here, we'll capture profiles for 100% of transactions.
  profilesSampleRate: 1.0,
  // ___PRODUCT_OPTION_END___ profiling
  // ___PRODUCT_OPTION_START___ session-replay
  // Record session replays for 100% of errors and 10% of sessions
  replaysOnErrorSampleRate: 1.0,
  replaysSessionSampleRate: 0.1,
  integrations: [Sentry.mobileReplayIntegration()],
  // ___PRODUCT_OPTION_END___ session-replay
});

function RootLayout() {
  return <Stack />;
}

export default Sentry.wrap(RootLayout);
```

### Wrap Your App

Wrap the root component of your application with `Sentry.wrap`:

```javascript {tabTitle:App.js or app/_layout.js}
export default Sentry.wrap(RootLayout);
```

### Add the Sentry Expo Plugin

To ensure bundles and source maps are automatically uploaded during the native applications builds, add `withSentry` to the Expo application configuration:

> <Include name="react-native-expo-plugin-code-snippet.mdx" />

Reviewer: I'm not seeing `withSentry` in the `app.json/app.config.json` snippet. Is that expected? 

<img width="1097" height="609" alt="Image" src="https://github.com/user-attachments/assets/ab7c3e97-695f-4d04-ad2e-28b5c15f2f54" />

Author: Updated with f9686c4c2
